frontend/src/pages/CompareV2.tsx

```diff
@@ -459,11 +555,18 @@ function CompareV2(props: CompareV2Props) {
   useEffect(() => {
     if (runs) {
       setSelectedIds(runs.map(r => r.run!.id!));
-      setParamsTableProps(getParamsTableProps(runs));
+    }
+  }, [runs]);
+
+  useEffect(() => {
+    if (runs) {
+      const selectedIdsSet = new Set(selectedIds);
+      const selectedRuns: ApiRunDetail[] = runs.filter(run => selectedIdsSet.has(run.run!.id!));
+      setParamsTableProps(getParamsTableProps(selectedRuns));
```

Comment:
The parameter setting is moved so they refresh when the run selection changes as well.

Reply:
Is it possible to merge this `useEffect` with the above `useEffect`?


```
  useEffect(() => {
    if (runs) {
      setSelectedIds(runs.map(r => r.run!.id!));
    }
  }, [runs]);
```

Reply:
I don't think so - since the above `useEffect` re-calculates the `selectedIds` when the run list changes while the below `useEffect` runs whenever `selectedIds` changes (it is included as a dependency), combining the two requires infinite refresh. We cannot remove that dependency in combining the `useEffect`s because the `selectedIds` are also changed within the `RunList` component.